components/NameplateGroup.tsx

```diff
@@ -3,70 +3,136 @@ import Shape, {Polygon} from '../components/Shape';
+  const shortLabelRef = useRef(null);
+  const longLabelRef = useRef(null);
 
-  useEffect(() => {
-    if (isFirstAnimation) {
-      setIsFirstAnimation(false);
-      return;
+  useEffect(animate
+    ? () => {
+      if (isFirstAnimation) {
+        setIsFirstAnimation(false);
+        return;
+      }
+
+      animatePolygon({
+        target: containerRef.current,
+        polygonNode: polygonRef.current,
+        forwards: isHovered,
+        shape,
+      });
+      animateNameplate({
+        target: shortLabelRef.current,
+        subtarget: longLabelRef.current,
+        forwards: isHovered,
+      });
     }
+    : () => null, [isHovered]);
 
-    animatePolygon({
-      target: containerRef.current,
-      polygonNode: polygonRef.current,
-      forwards: isHovered,
-      shape,
-    });
-    animateNameplate({
-      target: shortNameRef.current,
-      subtarget: longNameRef.current,
-      forwards: isHovered,
-    });
-  }, [isHovered]);
+  const className = `${styles.nameplate} ${highlight ? styles.highlight : ''} ${expanded ? styles.expanded : ''}`;
 
   return (
-    <div className={`${styles.nameplate} ${currentUser ? styles.currentUser : ''} ${expanded ? styles.expanded : ''}`} onMouseEnter={() => setIsHovered(true)} onMouseLeave={() => setIsHovered(false)}>
+    <div className={className} onMouseEnter={() => setIsHovered(true)} onMouseLeave={() => setIsHovered(false)}>
       <div className={styles['nameplate-container']} ref={containerRef}>
         <Shape polygon={shape} ref={polygonRef} />
       </div>
-      <div className={styles['nameplate-short']} ref={shortNameRef}>{name[0]}</div>
-      <div className={styles['nameplate-long']} ref={longNameRef}>{name}</div>
+      <div className={styles['nameplate-short']} ref={shortLabelRef}>{shortText}</div>
+      <div className={styles['nameplate-long']} ref={longLabelRef}>{longText}</div>
     </div>
   );
 }
 
-interface NameplateGroupProps {
-  names: NameplateProps[];
-  expandCurrentUser?: boolean;
+function Nameplate({
+  name,
+  shape,
+  highlight = false,
+  expanded = false,
+}: NameplateProps): JSX.Element {
+  return (
+    <AnimatedShape
+      shortText={name[0]}
+      longText={name}
+      shape={shape}
+      highlight={highlight}
+      expanded={expanded}
+    />
+  );
 }
 
-function NameplateGroup({names, expandCurrentUser = false}: NameplateGroupProps): JSX.Element {
+function Overflow({value}: OverflowProps): JSX.Element {
+  return (
+    <AnimatedShape
+      shortText={`+${value}`}
+      longText=""
+      shape="diamond"
+      highlight={false}
+      expanded={false}
+      animate={false}
+    />
+  );
+}
+
+
+function NameplateGroup({
+  names,
+  expandCurrentUser = false,
+  limit = null,
+}: NameplateGroupProps): JSX.Element {
+  limit = limit ?? names.length;
+
   return (
     <>
-      {names.map(passedProps =>
+      {names.slice(0, limit).map(passedProps =>
         <div key={`${passedProps.name}-${passedProps.shape}`}>
```

Comment:
this div seems like filler space.. why cant we just put the key on the react component itself?